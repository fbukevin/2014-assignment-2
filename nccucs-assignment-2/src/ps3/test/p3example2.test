# Operations testing that checks the functionality of AddNode, AddEdge,
# and their related observers.  Also, tests that unrelated graphs or 
# unconnected nodes do not effect path finding outcomes.  Finally, makes 
# sure that the abstraction invariant holds (no duplicate edges).

# create generic nodes
CreateNode f1 1
CreateNode f2 2
CreateNode d5 5
CreateNode e1 1
CreateGraph G
CreateGraph H

# test the separation of graphs
ListNodes G
AddNode G f1
AddNode G f2
ListNodes G
ListNodes H
AddNode H f1
AddNode H f2
ListNodes G
ListNodes H
FindPath G f1 -> f2

# test adding edges
AddEdge G f1 f2
AddEdge H f1 f2
ListChildren G f1
ListChildren G f2
ListChildren H f1
AddEdge G f2 f1
ListChildren H f2
AddEdge H f2 f1
ListChildren G f2
ListChildren H f2

# test pathfinding is different between graphs
AddNode G d5
AddNode H d5
AddEdge G d5 f1
FindPath G d5 -> f2
FindPath H d5 -> f2
AddEdge H d5 f2
FindPath G d5 -> f1
FindPath H d5 -> f1

# test paths are only found through edges and self-pointing
# edges are valid
AddNode G e1
FindPath G f1 -> e1
FindPath G e1 -> f1
AddEdge G d5 e1
AddEdge G f2 d5
AddEdge G f2 f2
AddEdge G e1 e1
FindPath G f1 -> e1
FindPath G e1 -> f1
FindPath G e1 -> e1

# final state check with observers
ListNodes G
ListNodes H
ListChildren G f1
ListChildren G f2
ListChildren G d5
ListChildren G e1
ListChildren H f1
ListChildren H f2
ListChildren H d5