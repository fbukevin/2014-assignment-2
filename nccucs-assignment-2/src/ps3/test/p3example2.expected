# Operations testing that checks the functionality of AddNode, AddEdge,
# and their related observers.  Also, tests that unrelated graphs or 
# unconnected nodes do not effect path finding outcomes.  Finally, makes 
# sure that the abstraction invariant holds (no duplicate edges).

# create generic nodes
created node f1 with cost 1
created node f2 with cost 2
created node d5 with cost 5
created node e1 with cost 1
created graph G
created graph H

# test the separation of graphs
G contains:
added node f1 to G
added node f2 to G
G contains: f1 f2
H contains:
added node f1 to H
added node f2 to H
G contains: f1 f2
H contains: f1 f2
no path found in G

# test adding edges
added edge from f1 to f2 in G
added edge from f1 to f2 in H
the children of f1 in G are: f2
the children of f2 in G are:
the children of f1 in H are: f2
added edge from f2 to f1 in G
the children of f2 in H are:
added edge from f2 to f1 in H
the children of f2 in G are: f1
the children of f2 in H are: f1

# test pathfinding is different between graphs
added node d5 to G
added node d5 to H
added edge from d5 to f1 in G
shortest path in G: d5 f1 f2
no path found in H
added edge from d5 to f2 in H
shortest path in G: d5 f1
shortest path in H: d5 f2 f1

# test paths are only found through edges and self-pointing
# edges are valid
added node e1 to G
no path found in G
no path found in G
added edge from d5 to e1 in G
added edge from f2 to d5 in G
added edge from f2 to f2 in G
added edge from e1 to e1 in G
shortest path in G: f1 f2 d5 e1
no path found in G
shortest path in G: e1

# final state check with observers
G contains: d5 e1 f1 f2
H contains: d5 f1 f2
the children of f1 in G are: f2
the children of f2 in G are: d5 f1 f2
the children of d5 in G are: e1 f1
the children of e1 in G are: e1
the children of f1 in H are: f2
the children of f2 in H are: f1
the children of d5 in H are: f2